### PR TITLE
feat: add qa-explain-behavior skill

### PR DESCRIPTION
**Pull request by TetianaKhomenko**
Answers a QA engineer's questions about product behavior — what features exist, how flows work, who can do what, what the edge cases are, and what is not implemented — by reading the codebase and translating it into tester language, never implementation detail.

Every other skill in the catalog acts on test artifacts (generate, improve, sync, automate). None answered "what does this product actually do?", which is the question that comes before writing a single test case. It slots in ahead of qa-thinking and qa-write-test-cases.

Ported from an internal skill and generalized off Rails: the stack section is now a cheat-sheet where Rails sits alongside Node/Nest, Django/FastAPI and Laravel, rather than being the assumed default.

Wired into the qa-process plugin, the testing-workflow routing registry, and the README tables, with a reciprocal cross-reference from qa-thinking.

Drive-by fix: qa-process was missing from .claude-plugin/marketplace.json despite having a full plugin.json and a README row, so nothing inside it was installable from the marketplace.

### CODE REVIEW

File: skills/qa-explain-behavior/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+- **Models** — only *validations*, *state machines*, *callbacks*, and *scopes with business meaning*. Not columns, associations, or types.
+- **Views, serializers, mailers** — what the user sees and receives.
+- **Background jobs** — what happens after the user-visible action: notifications, syncs, cleanups.
+- **Permission / policy classes** — who can do what.
+- **Feature flags and config** — what is gated or disabled.
+- **Tests** — they describe expected behavior in plain English; very useful.
+
+Skim. Look for **what happens**, not **how it's built**.
+
+Names differ by stack — controllers may be routers or handlers, policy classes may be guards or middleware, background jobs may be tasks or queues. Find the equivalent role rather than the exact filename.
```
Comment (DavertMik) [inline skills/qa-explain-behavior/SKILL.md]: remove this section fully as it is a codebase-specific things
simplfy it by saying:

Review codebase especially for

- business logic
- service and controller layers
- feature flags
- configuration
- authorization and permissions system

**Review by DavertMik — CHANGES_REQUESTED** [review]

File: skills/qa-explain-behavior/SKILL.md (hunk 1 above)
Comment (TetianaKhomenko) [reply skills/qa-explain-behavior/SKILL.md]: thank you! done

File: skills/qa-explain-behavior/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
+---
+name: qa-explain-behavior
+description: Answers QA questions about product behavior — what features exist, how user flows work, what business rules apply, what edge cases the system handles, and what is or is not implemented. Use whenever the user asks a question about WHAT the system does (features, flows, rules, conditions, who-can-do-what, when-does-X-happen, is-Y-supported). Trigger even when the user does not say "QA" — phrases like "what happens when…", "can a user…", "is it possible to…", "does the system…", "is X implemented", "what should I test for…", "what are the edge cases of…", or "walk me through the flow for…" all qualify. Do NOT trigger for questions that are clearly about implementation (architecture, database schema, code structure, performance, types, dependencies).
```
Comment (olexandr13) [inline skills/qa-explain-behavior/SKILL.md:3]: `Do NOT trigger for questions that are clearly about implementation (architecture, database schema, code structure, performance, types, dependencies).`

why? I thought this skill is especially helphul in such cases

File: skills/qa-explain-behavior/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+Answer a QA engineer who is testing the product. They care about **product behavior**, not implementation. Read the codebase like a manual and translate it into plain-language descriptions of what the system does.
+
+| What | File |
+|------|------|
+| Output shape per question type — feature, flow, yes/no, permissions, edge cases | [references/answer-shapes.md](references/answer-shapes.md) |
+| Worked answers, plus bad-vs-good contrasts | [references/examples.md](references/examples.md) |
+
+## The mindset shift
+
+The QA engineer is not going to read the code. They are going to click buttons, fill forms, and try to break things. The answer should help them know what to click, what to expect, and where the product might fail them. Describe the product like someone who has used it end-to-end.
```
Comment (olexandr13) [inline skills/qa-explain-behavior/SKILL.md]: `fail them`?

File: skills/qa-explain-behavior/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+A QA engineer's mental model is built from:
+
+- **Actors** — who is doing this: guest, member, admin, owner, integration.
+- **Actions** — what they do: click, submit, upload, invite.
+- **Outcomes** — what they see or what changes: page, message, email, status.
+- **Rules** — when it's allowed, when it's blocked, what limits apply.
+- **Edge cases** — empty state, very large input, duplicate, expired, offline.
```
Comment (olexandr13) [inline skills/qa-explain-behavior/SKILL.md:29]: I would like not to explain these items

File: skills/qa-explain-behavior/references/examples.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
+# Worked examples
```
Comment (olexandr13) [inline skills/qa-explain-behavior/references/examples.md:1]: remove this file completely (and link to it)

File: skills/qa-explain-behavior/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+## What to ignore
+
+Unless the user explicitly asks, never mention:
+
+- Database tables, columns, indexes, schema.
+- Class names, file paths, line numbers, module structure.
+- Framework terminology and internals — ORM concepts, callbacks, middleware, queues, dependencies.
+- Variable names, method names, parameters, types.
+- SQL, code snippets, regex.
```
Comment (olexandr13) [inline skills/qa-explain-behavior/SKILL.md:55]: why?

File: skills/qa-explain-behavior/SKILL.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+- **Models** — only *validations*, *state machines*, *callbacks*, and *scopes with business meaning*. Not columns, associations, or types.
+- **Views, serializers, mailers** — what the user sees and receives.
+- **Background jobs** — what happens after the user-visible action: notifications, syncs, cleanups.
+- **Tests** — they describe expected behavior in plain English; very useful.
+
+## What to ignore
+
+Unless the user explicitly asks, never mention:
+
+- Database tables, columns, indexes, schema.
```
Comment (olexandr13) [inline skills/qa-explain-behavior/SKILL.md:51]: ?

File: skills/qa-explain-behavior/SKILL.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+**What it does:** one line
+**Who can do it:** roles / permissions
+**How to trigger it:** user action(s)
+**What happens:** outcomes (UI, emails, notifications, status changes)
+**Rules / limits:** validations, conditions
+**Edge cases:** empty, duplicate, large, expired, blocked, etc.
+**Not implemented:** anything the QA might expect that does NOT exist
+```
+
+Skip sections that don't apply. Don't pad with "N/A".
```
Comment (olexandr13) [inline skills/qa-explain-behavior/SKILL.md]: this contradicts the statement below
`Surface what is missing`...

File: skills/qa-explain-behavior/SKILL.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+**Surface what is missing.** The most valuable thing a QA engineer can hear is "this is not implemented", "there is no UI for this — only the API supports it", or "this only works for paying accounts". Always look for gaps and call them out: a flag, a TODO, a commented-out path, an action reachable only via API.
+
+**Be honest about uncertainty.** If the codebase is ambiguous, say "I see X but I'm not sure whether Y" rather than guessing. A QA testing on a wrong assumption is worse than a QA who knows to verify.
+
+## Do not write code
+
+Never produce code blocks, schema, configuration snippets, or pseudocode unless the user explicitly asks ("show me the code", "give me the SQL"). The default response is prose and bullets only. When tempted to paste code "to be helpful", describe the behavior instead.
```
Comment (olexandr13) [inline skills/qa-explain-behavior/SKILL.md:95]: why not schemas?

File: skills/qa-explain-behavior/SKILL.md (hunk 2 above)
Comment (olexandr13) [inline skills/qa-explain-behavior/SKILL.md:3]: skill takes source code as a source of truth
but its not explicitly mentioned

I prefer to take source code + working app as source of truth. or lets explicitly specify only code.
what do you think?

File: skills/qa-explain-behavior/SKILL.md [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+
+When the product **is** a CLI or an API, the commands, flags, and endpoints a user types are product surface, not implementation — name them exactly, the same way you would name a button.
```
Comment (olexandr13) [inline skills/qa-explain-behavior/SKILL.md:97]: did not understand this point...

File: skills/qa-explain-behavior/SKILL.md (hunk 2 above)
Comment (olexandr13) [inline skills/qa-explain-behavior/SKILL.md:3]: I would like to add the the selection of detalization level.
E.g. I want the basic understanding of some feature or full description with all implementation details.

File: skills/qa-explain-behavior/references/answer-shapes.md [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+**Duplicate:** …
+**Too large / too many:** …
+**Wrong format:** …
+**Expired / stale:** …
+**Permissions:** …
+**Concurrent:** two people doing this at once …
+**External failure:** the integration is down / the upload fails …
+```
+
+If a case is genuinely unhandled, say so plainly — "nothing stops this" is a
```
Comment (olexandr13) [inline skills/qa-explain-behavior/references/answer-shapes.md]: don't understand

File: skills/qa-explain-behavior/references/answer-shapes.md [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+
+## Comparison — "what's the difference between X and Y"
+
+Two columns, same rows. Only rows where they actually differ.
+
+## Phrasing uncertainty
+
+Never guess silently. Use the shape "I can see X, but I couldn't confirm Y":
+
+- "The API accepts this, but I couldn't find a UI that reaches it."
```
Comment (olexandr13) [inline skills/qa-explain-behavior/references/answer-shapes.md:105]: all other items are FORMAT of answer, but here its EXAMPLES
could be misleading

**Review by olexandr13 — CHANGES_REQUESTED** [review]

File: skills/qa-explain-behavior/SKILL.md (hunk 2 above)
Comment (TetianaKhomenko) [reply skills/qa-explain-behavior/SKILL.md:3]: approved to use only source code

File: skills/qa-explain-behavior/SKILL.md (hunk 10 above)
Comment (TetianaKhomenko) [reply skills/qa-explain-behavior/SKILL.md:97]: I don't want to be overloaded with technical information as manual QA who does not understand code

File: skills/qa-explain-behavior/SKILL.md (hunk 3 above)
Comment (TetianaKhomenko) [reply skills/qa-explain-behavior/SKILL.md]: agree, it's better to simply say `fail`

File: skills/qa-explain-behavior/SKILL.md (hunk 2 above)
Comment (TetianaKhomenko) [reply skills/qa-explain-behavior/SKILL.md:3]: I don't want to be overloaded with technical information as manual QA who does not understand code

File: skills/qa-explain-behavior/SKILL.md (hunk 4 above)
Comment (TetianaKhomenko) [reply skills/qa-explain-behavior/SKILL.md:29]: why?

File: skills/qa-explain-behavior/SKILL.md (hunk 6 above)
Comment (TetianaKhomenko) [reply skills/qa-explain-behavior/SKILL.md:55]: idk

File: skills/qa-explain-behavior/SKILL.md (hunk 7 above)
Comment (TetianaKhomenko) [reply skills/qa-explain-behavior/SKILL.md:51]: I don't want to be overloaded with technical information as manual QA who does not understand code

File: skills/qa-explain-behavior/SKILL.md (hunk 9 above)
Comment (TetianaKhomenko) [reply skills/qa-explain-behavior/SKILL.md:95]: I don't want to be overloaded with technical information as manual QA who does not understand code

File: skills/qa-explain-behavior/references/examples.md (hunk 5 above)
Comment (TetianaKhomenko) [reply skills/qa-explain-behavior/references/examples.md:1]: why?

File: skills/qa-explain-behavior/references/answer-shapes.md (hunk 11 above)
Comment (TetianaKhomenko) [reply skills/qa-explain-behavior/references/answer-shapes.md]: yeah, maybe it's better to remove this

File: skills/qa-explain-behavior/SKILL.md (hunk 2 above)
Comment (TetianaKhomenko) [reply skills/qa-explain-behavior/SKILL.md]: sounds good

File: skills/qa-explain-behavior/SKILL.md (hunk 8 above)
Comment (TetianaKhomenko) [reply skills/qa-explain-behavior/SKILL.md]: ok, let's remove this

File: skills/qa-explain-behavior/SKILL.md (hunk 2 above)
Comment (TetianaKhomenko) [reply skills/qa-explain-behavior/SKILL.md:3]: the skill's spine is "behavior, not implementation" — **What to ignore** and **Do not write code** both enforce that. A "full implementation details" level deliberately lifts that ban, so it needs to be opt-in only and never reachable by accident. 

I don't wanna do this
